src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerService.cs

```diff
@@ -170,10 +174,57 @@ private List<string> GetSupportedColumnTypes(TableInfo tableInfo)
             return columnTypes;
         }
 
+        private TableViewModel GetTableViewModel(TableInfo tableInfo, List<string> schemas)
+        {
+            var tableViewModel = new TableViewModel();
+            // Schema
+            string schema;
+            if (tableInfo.IsNewTable)
+            {
+                schema = schemas.Contains("dbo") ? "dbo" : schemas[0];
+            }
+            else
+            {
+                schema = tableInfo.Schema;
+            }
+            tableViewModel.Schema.Value = schema;
+
+            // Table Name
+            if (!tableInfo.IsNewTable)
+            {
+                tableViewModel.Name.Value = tableInfo.Name;
+            }
+
+            // TODO: set other properties of the table
+            return tableViewModel;
+        }
+
+        private TableDesignerView GetDesignerViewInfo(TableInfo tableInfo)
+        {
+            // TODO: set the view information
+            var view = new TableDesignerView();
+            return view;
+        }
+
         private List<string> GetSchemas(TableInfo tableInfo)
         {
-            //TODO: get the schemas.
             var schemas = new List<string>();
+            using (var connection = new SqlConnection(tableInfo.ConnectionString))
```

Comment:
We should try to avoid having boilerplate code like this in each function that needs to make a query. I believe the ExecuteReader method in the ReliableConnectionHelper can be useful here - or if you can't find something similar that does what you need then I'd suggest making one that does and handles all the creating connection/opening/etc